packages/patternfly-4/react-core/src/components/Progress/Progress.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import styles from '@patternfly/patternfly-next/components/Progress/styles.css';
+import { css, getModifier } from '@patternfly/react-styles';
+import PropTypes from 'prop-types';
+import ProgressContainer, { ProgressMeasureLocation, ProgressVariant } from './ProgressContainer';
+
+export const ProgressSize = {
+  sm: 'sm',
+  md: 'md',
+  lg: 'lg'
+};
+
+const propTypes = {
+  /** Classname for progress component. */
+  className: PropTypes.string,
+  /** Size variant of progress. */
+  size: PropTypes.oneOf(Object.keys(ProgressSize)),
+  /** Where the measure percent will be located. */
+  measureLocation: PropTypes.oneOf(Object.keys(ProgressMeasureLocation)),
+  /** Status variant of progress. */
+  variant: PropTypes.oneOf(Object.keys(ProgressVariant)),
+  /** Title above progress. */
+  title: PropTypes.string,
+  /** Actual value of progress. */
+  value: PropTypes.number,
+  /** DOM id for progress component. */
+  id: PropTypes.string,
+  /** Minimal value of progress. */
+  min: PropTypes.number,
+  /** Maximum value of progress. */
+  max: PropTypes.number,
+  /** Dynamic description of progress. */
+  valueText: PropTypes.string
+};
+
+const defaultProps = {
+  className: '',
+  measureLocation: ProgressMeasureLocation.top,
+  variant: ProgressVariant.info,
+  id: `progress-${new Date().getTime()}${Math.random()
```

Comment:
If you have multiple progress components the generated ids will probably be the same.  I ran into this issue in the nav. Solution was to generate the id in the constructor, example: https://github.com/patternfly/patternfly-react/pull/626/files#diff-43517af4b617baa06579455d73f357d6R63

Reply:
The value in default props is evaluated when the default props object is created. Which is essentially on import, not when the component is created. It also only happens once. The constructor is called when an instance of the component is created. So it is called once for each instance this is rendered.